images/dind/release.sh

```diff
@@ -1,19 +1,44 @@
-set -ex
+#!/bin/bash
 
-# ./build.sh
+set -exo pipefail
+
+# Ensure working dir is clean
+git status
+if [[ -z $(git status -s) ]]
+then
+  echo "tree is clean"
+else
+  echo "tree is dirty, please commit changes before running this"
+  exit 1
+fi
+
+# This script should be run after its sibliing, build.sh, and
+# after any related tests have passed.
 
 # Match version with Docker version
 version=$(docker run --rm -v "$PWD":/app treeder/bump  --extract --input "`docker -v`")
 echo "Version: $version"
 M=$(docker run --rm treeder/bump --format M --input "$version")
 Mm=$(docker run --rm treeder/bump --format M.m --input "$version")
 
+# Calculate new release version
+DIND_NEW=$(echo "$DIND_PREV" | perl -pe 's/\d+\.\d+\.\K(\d+)/$1+1/e')
+
+# Add appropriate docker tags
 docker tag fnproject/dind:latest fnproject/dind:$version
 # be nice to have bump image do all of this tagging and pushing too (mount docker sock and do it all)
 docker tag fnproject/dind:$version fnproject/dind:$Mm
 docker tag fnproject/dind:$version fnproject/dind:$M
+docker tag fnproject/dind:$version fnproject/dind:release-$DIND_NEW
```

Comment:
Curious what this tag is for?

Reply:
Historical traceability / reproducability. You can find the previous DIND tag for any fnserver release and get hold of the image it was built atop of.

(Obviously you could also dive in and grab the docker layers directly; this is just more explicit.)